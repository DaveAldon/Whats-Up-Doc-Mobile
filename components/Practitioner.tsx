import React from "react";
import { Text, View } from "../components/Themed";
import { useTheme } from "@react-navigation/native";
import * as enums from "../constants/enums";
