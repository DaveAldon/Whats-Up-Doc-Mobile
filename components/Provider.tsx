import * as React from "react";
import { Linking, StyleSheet, TouchableOpacity } from "react-native";
import { Text, View } from "./Themed";
import * as enums from "../constants/enums";
import * as Device from "expo-device";

interface IPractitioner {
  name?: string;
  org?: string;
  orgID?: string;
  pID?: string;
  providerRole?: string;
  providerSpecialty?: string;
}

export const renderInner = (colors: any, practitioner: IPractitioner, providerData: any) => {
  const providerSpecific = {
    providerNPI: providerData?.identifier[0]?.value,
    phone: providerData?.telecom[1]?.value || "No phone number",
    email: providerData?.telecom[0]?.value || "No email",
  };
  return (
    <View style={[styles.panel, { backgroundColor: colors.border }]}>
      <View style={{ width: Device.modelName === "iPad" ? "50%" : "100%", backgroundColor: enums.colors.transparent }}>
        <Text style={styles.panelTitle}>{practitioner.name}</Text>
        <Text style={styles.panelSubtitle}>{practitioner.org}</Text>
        <Text style={styles.panelSubtitle}>Organization ID - {practitioner.orgID}</Text>

        <Text style={styles.panelSubtitle}>Provider NPI - {providerSpecific.providerNPI}</Text>

        <Text style={styles.panelSubtitle}>{practitioner.providerSpecialty}</Text>
        <Text style={styles.panelSubtitle}>Role ID - {practitioner.providerRole}</Text>

        <TouchableOpacity onPress={() => Linking.openURL(`mailto:${providerSpecific.email}`)}>
          <View style={[{ backgroundColor: colors.primary }, styles.panelButton]}>
            <Text style={styles.panelButtonTitle}>{providerSpecific.email}</Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity onPress={() => Linking.openURL(`tel:${providerSpecific.phone}`)}>
          <View style={[{ backgroundColor: colors.primary }, styles.panelButton]}>
            <Text style={styles.panelButtonTitle}>{providerSpecific.phone}</Text>
          </View>
        </TouchableOpacity>
      </View>
    </View>
  );
};

export const renderHeader = (colors: any) => (
  <View style={[{ backgroundColor: colors.border }, styles.header]}>
    <View style={[{ backgroundColor: colors.border }, styles.panelHeader]}>
      <View style={[{ backgroundColor: colors.text }, styles.panelHandle]} />
    </View>
  </View>
);

const styles = StyleSheet.create({
  header: {
    shadowColor: "#000000",
    paddingTop: 20,
    borderTopLeftRadius: 20,
    borderTopRightRadius: 20,
  },
  panelHeader: {
    alignItems: "center",
  },
  panelHandle: {
    width: 40,
    height: 8,
    borderRadius: 4,
    marginBottom: 10,
  },
  bottomSheet: {
    height: 600,
    padding: 20,
  },
  card: {
    height: 50,
    marginBottom: 10,
    borderRadius: 10,
    flexDirection: "row",
    alignItems: "center",
    paddingHorizontal: 20,
    justifyContent: "space-between",
  },
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
  },
  title: {
    fontSize: 20,
    fontWeight: "bold",
  },
  separator: {
    marginVertical: 30,
    height: 1,
    width: "80%",
  },
  panel: {
    height: 600,
    padding: 20,
    alignItems: "center",
  },
  panelTitle: {
    fontSize: 27,
    height: 35,
  },
  panelSubtitle: {
    fontSize: 14,
    color: "gray",
    height: 20,
    marginVertical: 5,
  },
  panelButton: {
    padding: 20,
    borderRadius: 10,
    alignItems: "center",
    marginVertical: 10,
  },
  panelButtonTitle: {
    fontSize: 14,
    fontWeight: "bold",
    color: "white",
  },
});
