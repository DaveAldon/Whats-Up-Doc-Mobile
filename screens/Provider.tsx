import * as React from "react";
import { StyleSheet } from "react-native";

import EditScreenInfo from "../components/EditScreenInfo";
import { Text, View } from "../components/Themed";

interface IProps {
  route: { params: any };
  navigation: any;
  response: any;
  error: any;
}

export default function Provider(props: IProps) {
  const { code } = props.route.params;
  //console.log(await FHIR.GetProviders(disease.code));
  return (
    <View style={styles.container}>
      <Text style={styles.title}>{code}</Text>
      <View style={styles.separator} lightColor="#eee" darkColor="rgba(255,255,255,0.1)" />
      <EditScreenInfo path="/screens/TabTwoScreen.js" />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
  },
  title: {
    fontSize: 20,
    fontWeight: "bold",
  },
  separator: {
    marginVertical: 30,
    height: 1,
    width: "80%",
  },
});
