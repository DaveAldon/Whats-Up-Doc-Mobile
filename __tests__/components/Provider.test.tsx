import React from "react";
import renderer from "react-test-renderer";
import { RenderInner, RenderHeader } from "../../components/Provider";

describe("Provider <RenderInner />", () => {
  const mockProvider = {
    colors: "colors",
    practitioner: "practitioner",
    providerData: {
      identifier: ["value"],
      telecom: ["value", "value"],
    },
  };
  beforeEach(() => {
    jest.useFakeTimers();
  });

  it("has children", () => {
    const tree = renderer.create(<RenderInner {...mockProvider} />).toJSON();
    expect(tree?.children.length).toBeGreaterThan(0);
  });

  it("renders correctly", () => {
    const tree = renderer.create(<RenderInner {...mockProvider} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Provider <RenderHeader />", () => {
  const mockColor = "colors";
  beforeEach(() => {
    jest.useFakeTimers();
  });

  it("has children", () => {
    const tree = renderer.create(<RenderHeader {...mockColor} />).toJSON();
    expect(tree?.children.length).toBeGreaterThan(0);
  });

  it("renders correctly", () => {
    const tree = renderer.create(<RenderHeader {...mockColor} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
