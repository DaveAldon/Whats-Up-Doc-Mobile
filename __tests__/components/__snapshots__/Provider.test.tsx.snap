// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Provider <RenderHeader /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Array [
        Object {
          "backgroundColor": undefined,
        },
        Object {
          "borderTopLeftRadius": 20,
          "borderTopRightRadius": 20,
          "paddingTop": 20,
          "shadowColor": "#000000",
        },
      ],
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Array [
          Object {
            "backgroundColor": undefined,
          },
          Object {
            "alignItems": "center",
          },
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
          },
          Array [
            Object {
              "backgroundColor": undefined,
            },
            Object {
              "borderRadius": 4,
              "height": 8,
              "marginBottom": 10,
              "width": 40,
            },
          ],
        ]
      }
    />
  </View>
</View>
`;

exports[`Provider <RenderInner /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Array [
        Object {
          "alignItems": "center",
          "height": 600,
          "padding": 20,
        },
        Object {
          "backgroundColor": undefined,
        },
      ],
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "backgroundColor": "transparent",
          "width": "100%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
          },
          Object {
            "fontSize": 27,
            "height": 35,
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#000",
          },
          Object {
            "color": "gray",
            "fontSize": 14,
            "height": 20,
            "marginVertical": 5,
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#000",
          },
          Object {
            "color": "gray",
            "fontSize": 14,
            "height": 20,
            "marginVertical": 5,
          },
        ]
      }
    >
      Organization ID - 
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000",
          },
          Object {
            "color": "gray",
            "fontSize": 14,
            "height": 20,
            "marginVertical": 5,
          },
        ]
      }
    >
      Provider NPI - 
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000",
          },
          Object {
            "color": "gray",
            "fontSize": 14,
            "height": 20,
            "marginVertical": 5,
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#000",
          },
          Object {
            "color": "gray",
            "fontSize": 14,
            "height": 20,
            "marginVertical": 5,
          },
        ]
      }
    >
      Role ID - 
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
            },
            Array [
              Object {
                "backgroundColor": undefined,
              },
              Object {
                "alignItems": "center",
                "borderRadius": 10,
                "marginVertical": 10,
                "padding": 20,
              },
            ],
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
              },
              Object {
                "color": "white",
                "fontSize": 14,
                "fontWeight": "bold",
              },
            ]
          }
        >
          No email
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
            },
            Array [
              Object {
                "backgroundColor": undefined,
              },
              Object {
                "alignItems": "center",
                "borderRadius": 10,
                "marginVertical": 10,
                "padding": 20,
              },
            ],
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
              },
              Object {
                "color": "white",
                "fontSize": 14,
                "fontWeight": "bold",
              },
            ]
          }
        >
          No phone number
        </Text>
      </View>
    </View>
  </View>
</View>
`;
